applications/tari_base_node/src/command_handler.rs

```diff
@@ -189,18 +190,8 @@ impl CommandHandler {
 
     /// Function to process the get-state-info command
     pub fn state_info(&self) {
-        let mut channel = self.state_machine_info.clone();
-        self.executor.spawn(async move {
-            match channel.recv().await {
-                None => {
-                    info!(
-                        target: LOG_TARGET,
-                        "Error communicating with state machine, channel could have been closed"
-                    );
-                },
-                Some(data) => println!("Current state machine state:\n{}", data),
-            };
-        });
+        let watch = self.state_machine_info.clone();
+        println!("Current state machine state:\n{}", *watch.borrow());
```

Comment:
Nice find

Reply:
👍  The new tokio watch api makes it much more obvious when you are waiting for a change vs just getting the value - which is nice.